scipy/sparse/_construct.py

```diff
@@ -439,6 +471,11 @@ def eye(m, n=None, k=0, dtype=float, format=None):
         You are encouraged to use ``eye_array`` to take advantage
         of the sparse array functionality.
 
+    Returns
```

Comment:
Let's move this up before the warning section

Reply:
I've moved the warning sections, rechecked the single/double backticks and tried to expend those backtick suggestions to the rest of this module. 
@perimosocordiae this should be ready for a re-review. :)